### PR TITLE
Add astype for DataArray and DataProxy

### LINKED ISSUE #910: Add `astype` for data arrays
This was previously added for variables. How should this behave for data arrays --- convert also coords, or keep them? Maybe worth checking what http://xarray.pydata.org/en/stable/generated/xarray.DataArray.astype.html does.

I suppose keeping coords untouched is maybe the best and most obvious choice, and keeps things simple (separation of concerns)?

For now this will be required in particular for benchmarking purposes, i.e., for running workflows in different precision.

### PR DESCRIPTION
**Pull request by DanNixon**
Fixes #910

### CODE REVIEW

File: python/dataset.cpp [hunk 1]
```diff
@@ -635,6 +648,9 @@ void init_dataset(py::module &m) {
         :return: A new variable that contains the union of all masks.
         :rtype: Variable)");
 
+  bind_astype(dataArray);
+  bind_astype(dataProxy);
```
Comment (SimonHeybrock) [inline python/dataset.cpp]: Do we need both? `DataArray` is implicitly convertible to `DataProxy`.

Comment (DanNixon) [reply python/dataset.cpp]: Yes, as this is adding `astype()` as a member of `DataArray` and `DataProxy`, not just a free function  of `scipp`.

Comment (SimonHeybrock) [reply python/dataset.cpp]: :man_facepalming: :+1:

**Comment by SimonHeybrock** [comment]
@DanNixon There are some conflicts now.

**Pull request merged**